contracts/libraries/VaultLifecycleTreasury.sol

```diff
@@ -0,0 +1,796 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+import {SafeMath} from "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {Vault} from "./Vault.sol";
+import {ShareMath} from "./ShareMath.sol";
+import {IStrikeSelection} from "../interfaces/IRibbon.sol";
+import {GnosisAuction} from "./GnosisAuction.sol";
+import {
+    IOtokenFactory,
+    IOtoken,
+    IController,
+    GammaTypes
+} from "../interfaces/GammaInterface.sol";
+import {IERC20Detailed} from "../interfaces/IERC20Detailed.sol";
+import {IGnosisAuction} from "../interfaces/IGnosisAuction.sol";
+import {SupportsNonCompliantERC20} from "./SupportsNonCompliantERC20.sol";
+import {UniswapRouter} from "./UniswapRouter.sol";
+
+library VaultLifecycleTreasury {
+    using SafeMath for uint256;
+    using SupportsNonCompliantERC20 for IERC20;
```

Comment:
The Open Zeppelin audit says this only supports USDT?
Here it looks like we are using this for collateral assets?

https://blog.openzeppelin.com/ribbon-finance-audit/

Reply:
Yes, seems like. I'm not too sure about this